docs/pages/management/dynamic-resources/terraform-provider.mdx

```diff
@@ -83,66 +79,60 @@ To prepare credentials for a local Teleport user:
            - access_list
            verbs: ['list','create','read','update','delete']
    version: v7
-   ---
-   kind: user
-   metadata:
-     name: terraform
-   spec:
-     roles: ['terraform']
-   version: v2
    ```
 
-   These settings configure a user and role named `terraform` with the permissions
+   These settings configure a role named `terraform` with the permissions
    required to manage resources in your Teleport cluster.
 
-1. Create the `terraform` user and role by running the following command:
+1. Create the `terraform` role by running the following command:
 
    ```code
-   $ tctl create -f terraform.yaml
+   $ tctl create -f terraform_role.yaml
    role 'terraform' has been created
-   user "terraform" has been created
    ```
 
-   The `terraform` user can't sign in to get credentials, so you must have another user
-   **impersonate** the `terraform` account to request a certificate.
-
-1. Create a new file called `terraform-impersonator.yaml` and open it in an editor.
+1. We're going to create two scripts that will let us quickly and repeatably create and destroy
+   our local Terraform bot. First, create a new file called `create_and_run_terraform_bot.sh`,
+   open it in an editor, and paste in the following content:
+
+   ```shell
+   #! /usr/bin/env bash
+
+   TOKEN=$(tctl bots add terraform-test --roles=terraform --format=json | jq -r '.token_id')
+   tbot start \
+     --destination-dir=./terraform-identity \
+     --token="$TOKEN" \
+     # Replace with the host and port of your Teleport Auth Service
+     # or Teleport Proxy Service
+     --auth-server=tele.example.com:443 \
+     --join-method=token
+   ```
 
-1. Configure a role that enables your user to impersonate the Terraform user by pasting
-the following content into the `terraform-impersonator.yaml` file:
+   When run, this creates a new bot with the token retrieved from the JSON output. That token
+   is piped into the start command. For security reasons, we don't allow bot join tokens to be
```

Comment:
> For security reasons, we don't allow bot join tokens to be created in any way other than adding a bot

I'm not sure this is true - if anything it's an awkward edge in the UX experience at the moment that should be fixed in coming months.

Reply:
Ok, I remember seeing/hearing something that it was for security. Maybe that was Ben making it up too hah.